app/views/handlers/list_collector.py

```diff
@@ -55,5 +69,22 @@ def handle_post(self):
             # wtforms Form parents are not discoverable in the 2.3.3 implementation
             self.questionnaire_store_updater.update_answers(self.form.data)
             self.questionnaire_store_updater.save()
-        else:
+        elif not self.rendered_block["repeating_blocks"] or (self.rendered_block["repeating_blocks"] and self._is_list_collector_complete()):
             return super().handle_post()
+
+    def _is_list_collector_complete(self):
```

Comment:
This would return false if the schema didn't have repeating blocks?

Reply:
So this is why I added the add block as part of list item progress, so list item completeness and then list collector completeness could all be evaluated in the same way via `is_section_complete` for list items, and then this method for the list collector itself, and there's no additional handle case for whether or not there is repeating blocks.

Reply:
I'm not sure if I follow but it seems like we are doing things we don't need to do. For example, we are adding the add block to the completed blocks just so this check passes when we don't have repeating blocks, even though the check is redundant in the scenario. We should avoid checking the store if we don't need to.

We could simplify this to not check the store if not needed. Plus in turn, you can simply your elif logic in handle_post.
```py
	@cached_property
    def repeating_blocks(self) -> list[dict] | None:
        return self.rendered_block.get("repeating_blocks")

    def _is_list_collector_complete(self) -> bool:
        if not self.repeating_blocks:
            return True

        return bool(self._get_first_incomplete_repeating_location())
```

Then handle post elif would become:

```py
        elif self._is_list_collector_complete():
            return super().handle_post()
```